### ISSUE TITLE
Arrow key behaviour inconsistent between widget areas

### ISSUE THREAD

**Issue by tellthemachines** [WordPress/gutenberg#38264]
### Description

When navigating inside widget areas with the arrow keys, Arrow Down doesn't always focus the "Add block" button at the very bottom of each widget area. 

Testing repeatedly yields varying results: in my testing, on macOS with Chrome, Safari and Firefox, most often the "Add block" button is skipped in the active widget areas, and focused in the inactive widgets area, but @joedolson reported different behaviour [here](https://wordpress.slack.com/archives/C02RP4X03/p1643054055395100?thread_ts=1642790310.369500&cid=C02RP4X03)

Arrow Right often works to focus the button when Arrow Down doesn't. Arrow Up, when navigating in the opposite direction, also works most of the time.

### Step-by-step reproduction instructions

1. In the Widgets editor at `wp-admin/widgets.php`, focus the first widget area, and use the Arrow Down key to traverse it from top to bottom.
2. Notice if "Add block" button at the bottom of the area gets focus or not.
3. Repeat for all widget areas, including inactive widgets.

### Environment info

macOS Monterey, on Chrome, Safari and Firefox.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by noisysocks** [comment]
Are there other areas where we always show the block appender at the end of a block editor? I think not, but thought I'd double check.
